### PR TITLE
Add Code For Binary Coded String and update README

### PR DESCRIPTION
**Pull request by ANUJAGRAWAL2000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes
### Description

(Write your answer here.)

### Checklist

- [Yes ] I've read the contribution guidelines.
- [Yes ] I've checked the issue list before deciding what to submit.
- [Yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#4541
(Write your answer here.)

### CODE REVIEW

**Comment by ANUJAGRAWAL2000** [comment]
@HarshCasper I created this PR for issue #4541 Please check this one and merged this PR.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ANUJAGRAWAL2000 Add space and time complexity.

**Comment by ANUJAGRAWAL2000** [comment]
@iamrajiv and @HarshCasper I have make the requested changes please merge it.

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                Z++;
+            A[Z]=1;
+        }
+        for(i=0;i<Y;i++)
+        {
+            if(A[i]==0)
+                return false;
+        }
+        return true;
+    }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Remove one tab space

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+/*
+Time Complexity=O(N)
+Space Complexity=O(2^K)
+where N is the length of the given string
+and K is the given integer.
+
+Sample Input Output
+Input: s = "00110", k = 2
+Output: true
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Rewrite it like that as your code produces like that.

Capture

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Constraints:
+1 <= s.length <= 5 * 10^5
+s consists of 0's and 1's only.
+1 <= k <= 20
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+bool hasAllCodes(string s, int k) {
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one line description of your function above the function

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    int X,Y,Z,i;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Don't use single letters for variable use proper naming conventions.

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    X=s.length();
+    Y=pow(2,k);
+    int A[2022600]={0};
+    Z=0;
+    for(i=0;i<k;i++)
+    {
+        if(i!=0  &&  Z!=0)
+        Z=(Z<<1);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        if(s[i]=='1')
+        Z++;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    A[Z]=1;
+    for(i=k;i<X;i++)
+    {
+        if(Z>=(pow(2,k-1)))
+        Z-=(pow(2,k-1));
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        if(Z!=0)
+        Z=(Z<<1);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+        if(s[i]=='1')
+        Z++;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        A[Z]=1;
+    }
+    for(i=0;i<Y;i++)
+    {
+        if(A[i]==0)
+        return false;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    }
+    return true;
+}
+int main()
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp:56]: Add a new line above main

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+{
+    string S;
+    int K;
+    cout<<"Enter an String"<<endl;
+    cin>>S;
+    cout<<"Enter the integer"<<endl;
+    cin>>K;
+    bool Ans=hasAllCodes(S,K);
+    if(Ans==true)
+    cout<<"Yes"<<endl;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    else
+    cout<<"No"<<endl;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp]: Add a one tab space inside body of condition

File: C-Plus-Plus/cp/Binary Coded String.cpp [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+Sample Input Output
+Input:
+Enter an String
+00110
+Enter the integer
+2
+Output:
+Yes
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Binary Coded String.cpp:86]: Add the new line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**